{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "KerberosAuthenticator.java",
  "functionName": "wrapExceptionWithMessage",
  "functionId": "wrapExceptionWithMessage___exception-T__msg-String",
  "sourceFilePath": "hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/client/KerberosAuthenticator.java",
  "functionStartLine": 226,
  "functionEndLine": 239,
  "numCommitsSeen": 23,
  "timeTaken": 827,
  "changeHistory": [
    "b0d3c877e30312820124cac2eff737fddac9e484"
  ],
  "changeHistoryShort": {
    "b0d3c877e30312820124cac2eff737fddac9e484": "Yintroduced"
  },
  "changeHistoryDetails": {
    "b0d3c877e30312820124cac2eff737fddac9e484": {
      "type": "Yintroduced",
      "commitMessage": "HADOOP-12897. KerberosAuthenticator.authenticate to include URL on IO failures. Contributed by Ajay Kumar.\n",
      "commitDate": "20/02/18 6:18 PM",
      "commitName": "b0d3c877e30312820124cac2eff737fddac9e484",
      "commitAuthor": "Arpit Agarwal",
      "diff": "@@ -0,0 +1,14 @@\n+   static \u003cT extends Exception\u003e T wrapExceptionWithMessage(\n+      T exception, String msg) {\n+    Class\u003c? extends Throwable\u003e exceptionClass \u003d exception.getClass();\n+    try {\n+      Constructor\u003c? extends Throwable\u003e ctor \u003d exceptionClass\n+          .getConstructor(String.class);\n+      Throwable t \u003d ctor.newInstance(msg);\n+      return (T) (t.initCause(exception));\n+    } catch (Throwable e) {\n+      LOG.debug(\"Unable to wrap exception of type {}, it has \"\n+          + \"no (String) constructor.\", exceptionClass, e);\n+      return exception;\n+    }\n+  }\n\\ No newline at end of file\n",
      "actualSource": "   static \u003cT extends Exception\u003e T wrapExceptionWithMessage(\n      T exception, String msg) {\n    Class\u003c? extends Throwable\u003e exceptionClass \u003d exception.getClass();\n    try {\n      Constructor\u003c? extends Throwable\u003e ctor \u003d exceptionClass\n          .getConstructor(String.class);\n      Throwable t \u003d ctor.newInstance(msg);\n      return (T) (t.initCause(exception));\n    } catch (Throwable e) {\n      LOG.debug(\"Unable to wrap exception of type {}, it has \"\n          + \"no (String) constructor.\", exceptionClass, e);\n      return exception;\n    }\n  }",
      "path": "hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/client/KerberosAuthenticator.java"
    }
  }
}